kernel/interrupts/src/aarch64/mod.rs

```diff
@@ -117,23 +132,29 @@ pub fn init() -> Result<(), &'static str> {
         inner.set_minimum_priority(0);
         *gic = Some(inner);
 
-        log::info!("Done Configuring the GIC");
+        info!("Done Configuring the GIC");
 
         Ok(())
     }
 }
 
-pub fn enable_timer_interrupts(enable: bool) -> Result<(), &'static str> {
-    // called everytime the timer ticks.
-    extern "C" fn timer_handler(_exc: &ExceptionContext) -> bool {
-        info!("timer int!");
-        loop {}
-    }
-
-    // register the handler for the timer IRQ.
-    if let Err(existing_handler) = register_interrupt(CPU_LOCAL_TIMER_IRQ, timer_handler) {
-        if timer_handler as *const HandlerFunc != existing_handler {
-            return Err("An interrupt handler has already been setup for the timer IRQ number");
+/// This function registers an interrupt handler for the CPU-local
+/// timer, enables the routing of this interrupt in the GIC, and
+/// turns the timer on, when `enable` is true.
+///
+/// When `enable` is false, the handler is deregistered, routing is
+/// disabled in the GIC, and the timer is turned off.
+pub fn enable_timer_interrupts(enable: bool, timer_tick_handler: HandlerFunc) -> Result<(), &'static str> {
```

Comment:
this function does too many things. I think it's supposed to be a counterpart to `LocalApic::enable_lvt_timer()`, but it includes both initialization, registration, and enabling.

It should be possible to initialize and register an interrupt handler in one function, and then enable/disable it in a different function. This is required for `preemption`, among other crates.